src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultConverters.cs

```diff
@@ -2,75 +2,68 @@
-            new JsonValueConverterDoubleNullable(),
-            new JsonValueConverterDecimalNullable(),
-            new JsonValueConverterDateTimeNullable(),
-            null,
-            new JsonValueConverterString()
-        };
-
-        internal static object GetDefaultPropertyValueConverter(Type propertyType, bool isNullable)
-        {
-            object converter = null;
-
-            int typeCode = (int)Type.GetTypeCode(propertyType);
-            if (typeCode <= MaxTypeCode)
+            if (type.IsEnum)
             {
-                if (isNullable)
-                {
-                    converter = s_NullableConverters[typeCode];
-                }
-                else
-                {
-                    converter = s_Converters[typeCode];
-                }
+                return Activator.CreateInstance(
+                    typeof(JsonValueConverterEnum<>).MakeGenericType(type),
+                    BindingFlags.Instance | BindingFlags.NonPublic,
+                    binder: null,
+                    new object[] { false },
+                    culture: null);
             }
-
-            return converter;
+            else if (type == typeof(Enum))
+            {
+                return new JsonValueConverterEnum(false);
+            }
+            TypeCode typeCode = Type.GetTypeCode(type);
+            switch (typeCode)
+            {
+                case TypeCode.Boolean:
+                    return new JsonValueConverterBoolean();
+                case TypeCode.Char:
+                    return new JsonValueConverterChar();
+                case TypeCode.SByte:
+                    return new JsonValueConverterSByte();
+                case TypeCode.Byte:
+                    return new JsonValueConverterByte();
+                case TypeCode.Int16:
+                    return new JsonValueConverterInt16();
+                case TypeCode.UInt16:
+                    return new JsonValueConverterUInt16();
+                case TypeCode.Int32:
+                    return new JsonValueConverterInt32();
+                case TypeCode.UInt32:
+                    return new JsonValueConverterUInt32();
+                case TypeCode.Int64:
+                    return new JsonValueConverterInt64();
+                case TypeCode.UInt64:
+                    return new JsonValueConverterUInt64();
+                case TypeCode.Single:
+                    return new JsonValueConverterSingle();
+                case TypeCode.Double:
+                    return new JsonValueConverterDouble();
+                case TypeCode.Decimal:
+                    return new JsonValueConverterDecimal();
+                case TypeCode.DateTime:
+                    return new JsonValueConverterDateTime();
+                case TypeCode.String:
+                    return new JsonValueConverterString();
+            }
+            return null;
         }
     }
+
+    internal static class DefaultConverters<TValue>
```

Comment:
@steveharter Should this also be moved into a separate .cs file?

Reply:
Ideally, yes. Basically all classes.